Source/WebCore/css/StyleProperties.cpp

```diff
@@ -1355,6 +1354,38 @@ String StyleProperties::pageBreakPropertyValue(const StylePropertyShorthand& sho
     }
 }
 
+String StyleProperties::columnBreakPropertyValue(const StylePropertyShorthand& shorthand) const
+{
+    ASSERT(shorthand.length() == 1);
+    auto value = getPropertyCSSValue(shorthand.properties()[0]);
+    if (!value)
+        return String();
+    // FIXME: Remove this isCSSWideKeyword check after we do this consistently for all shorthands in getPropertyValue.
+    if (value->isCSSWideKeyword())
+        return value->cssText();
+
+    if (!is<CSSPrimitiveValue>(*value))
+        return String();
+
+    CSSValueID valueId = downcast<CSSPrimitiveValue>(*value).valueID();
+    switch (valueId) {
+    case CSSValueColumn:
+        return "always"_s;
+    case CSSValueAvoidColumn:
+        if (shorthand.id() != CSSPropertyWebkitColumnBreakInside)
+            return "avoid"_s;
+        return String();
+    case CSSValueAvoid:
+        if (shorthand.id() == CSSPropertyWebkitColumnBreakInside)
+            return value->cssText();
+        return String();
+    case CSSValueAuto:
+        return value->cssText();
```

Comment:
I'd prefer splitting inside to a separate function, given you can't end up with CSSValueColumn in break-inside.

Also, `emptyString()`/`nameString(*valueID)` would be preferable to `String()`/`cssText()`.

Reply:
It can't be `emptyString()` because [`asTextInternal()`](https://searchfox.org/wubkat/rev/fd9594cdf418745718908fd1869a116307a46e7a/Source/WebCore/css/StyleProperties.cpp#1756) checks `isNull()`, and possibly other places too. I can use `emptyString()`, though. But note `String()` is used all over the place.

I could use `nameString(CSSValueAuto)` but this wouldn't use the `cssTextCache`, not sure if that's bad.

Reply:
`cssText()` does seem to use `nameString()` underneath, so I guess either is fine. 

`nullString()` could work instead of `emptyString()`, not sure what's appropriate in your case.

Reply:
Though I guess you won't have access to the value by using `propertyAsValueID()` so, you'll likely need to use `nameString()`. I think bypassing the cache is fine in this case and looking up the string directly in the table is fine. We already do it for font.